scrapy/commands/crawl.py

```diff
@@ -54,8 +54,14 @@ def run(self, args, opts):
             raise UsageError("running 'scrapy crawl' with more than one spider is no longer supported")
         spname = args[0]
 
-        self.crawler_process.crawl(spname, **opts.spargs)
-        self.crawler_process.start()
-
-        if self.crawler_process.bootstrap_failed:
+        res = self.crawler_process.crawl(spname, **opts.spargs)
+        
+        if hasattr(res, 'result') and res.result is not None and issubclass(res.result.type, Exception):
```

Comment:
:lipstick: I think we can make this shorter:

```suggestion
        if getattr(res, 'result') is not None and issubclass(res.result.type, Exception):
```

Reply:
cool, I'll use this shorter code

Reply:
I found this shorter code, it will cause `AttributeError: ‘Deferred’ object has no attribute ‘result’` if no exception is raised so I'll change it back

Reply:
My bad, you need to add a third parameter with the default value to avoid that: `getattr(res, 'result', None)`

Reply:
no problem, I've modified it, thanks